src/codecoverage/bot/code_coverage_bot/codecov.py

```diff
@@ -144,7 +144,12 @@ def go_from_trigger_mozilla_central(self):
             token=secrets[secrets.COVERALLS_TOKEN]
         )
         logger.info('Report generated successfully')
-
+        json_output = json.loads(output)
+        file_list = [file['name'] for file in json_output['source_files']]
+        expected_extensions = ['.js', '.cpp']
+        for extension in expected_extensions:
+            assert any([file.endswith(extension) for file in file_list]), 'No {} file in the generated report'.format(
```

Comment:
`any` should also work with iterators, so you don't need to create another list here.

any([val for val in [False, True]]) and any(val for val in [False, True]) should both work, but the second will use less memory (as you won't need to create a list in memory and then iterate it).